executor/builder.go

```diff
@@ -820,12 +821,81 @@ func (b *executorBuilder) buildHashJoin(v *plan.PhysicalHashJoin) Executor {
 	return e
 }
 
+// buildProjBelowAgg builds a ProjectionExec below AggregationExec.
+// If all the args of `aggFuncs`, and all the item of `groupByItems`
+// are columns or constants, we do not need to build the `proj`.
+func (b *executorBuilder) buildProjBelowAgg(aggFuncs []*aggregation.AggFuncDesc, groupByItems []expression.Expression, src Executor) Executor {
+	getScalarFunc := false
+	for i := 0; !getScalarFunc && i < len(aggFuncs); i++ {
+		f := aggFuncs[i]
+		for _, arg := range f.Args {
```

Comment:
how about:
```go
for j :=0; !getScalarFunc && j < len(f.Args); j++ {
    _, isScalarFunc := f.Args[j].(*expression.ScalarFunction);
    getScalarFunc = getScalarFunc || isScarlarFunc
}
```

Reply:
It seems this will not be better?

Reply:
This is not about performance. It will make code simper and reduce some `break` and `if`s.